src/preset.ts

```diff
@@ -988,12 +997,129 @@ async function expandConfigurePresetImpl(folder: string, name: string, workspace
+    // that supports the specified toolset.
+    if (!vsInstall) {
+        // sort VS installs in order of descending version. This ensures we choose the latest supported install first.
+        vsInstalls.sort((a, b) => {
+            if (a.isPrerelease && !b.isPrerelease) {
+                return 1;
+            } else if (!a.isPrerelease && b.isPrerelease) {
+                return -1;
+            }
+            return -compareVersions(a.installationVersion, b.installationVersion);
+        });
+
+        for (const vs of vsInstalls) {
+            // Check for existence of vcvars script to determine whether desired host/target architecture is supported.
+            // toolset.host will be set by getToolset.
+            if (await getVcVarsBatScript(vs, toolset.host!, arch)) {
+                // If a toolset version is specified then check to make sure this vs instance has it installed.
+                if (toolset.version) {
+                    const availableToolsets = await enumerateMsvcToolsets(vs.installationPath, vs.installationVersion);
+                    // forcing non-null due to false positive (toolset.version is checked in conditional)
+                    if (availableToolsets?.find(t => t.startsWith(toolset.version!))) {
+                        vsInstall = vs;
+                        break;
+                    }
+                } else if (!vsGeneratorVersion || vs.installationVersion.startsWith(vsGeneratorVersion.toString())) {
+                    // If no toolset version specified then choose the latest VS instance for the given generator
+                    vsInstall = vs;
+                    break;
+                }
+            }
+        }
+    }
+
+    if (!vsInstall) {
+        if (opts.compilerName) {
+            log.error(localize('specified.cl.not.found',
+                "Configure preset {0}: Compiler {1} with toolset {2} and architecture {3} was not found, you may need to run the 'CMake: Scan for Compilers' command if this toolset exists on your computer.",
+                opts.preset.name, `"${opts.compilerName}.exe"`, toolset.version ? `"${toolset.version},${toolset.host}"` : `"${toolset.host}"`, `"${arch}"`));
+        } else {
+            log.error(localize('vs.not.found', "Configure preset {0}: No Visual Studio installation found that supports the specified toolset {1} and architecture {2}, you may need to run the 'CMake: Scan for Compilers' command if this toolset exists on your computer.",
+                opts.preset.name, toolset.version ? `"${toolset.version},${toolset.host}"` : `"${toolset.host}"`, `"${arch}"`));
+        }
+    } else {
+        log.info(localize('using.vs.instance', "Using developer environment from Visual Studio (instance {0}, version {1}, installed at {2})", vsInstall.instanceId, vsInstall.installationVersion, `"${vsInstall.installationPath}"`));
+        const vsEnv = await varsForVSInstallation(vsInstall, toolset.host!, arch, toolset.version);
```

Comment:
This omits `VCToolsInstallDir`, which is a problem because that's how LLVM detects a VS2017+ toolset: https://github.com/llvm/llvm-project/blob/a288d8dad3b3f459bce3f908cd564f2963060642/llvm/lib/WindowsDriver/MSVCPaths.cpp#L512

See `visualStudio.ts`:
```cpp
//'VCToolsInstallDir', - disabled temporarily as it breaks downlevel toolset selection
```

Reply:
Yes, I'm simultaneously investigating that for this bug, if I can confirm that it won't break any scenarios, we can likely include this in this PR.

Reply:
@tristanlabelle I updated it, if you could please test this out in your scenarios that would be great.

Reply:
It works!! 😍